File: dathighlight/src/main/java/com/dat/dathighlight/position/OnTopPosCallback.java
package com.dat.dathighlight.position;

import android.graphics.RectF;

import com.dat.dathighlight.HighLight;

/**
 * Created by caizepeng on 16/8/20.
 */
public class OnTopPosCallback extends OnBaseCallback {
    public OnTopPosCallback() {
    }

    public OnTopPosCallback(float offset) {
        super(offset);
    }

    @Override
    public void getPosition(float rightMargin, float bottomMargin, RectF rectF, HighLight.MarginInfo marginInfo) {
        marginInfo.leftMargin = rectF.right - rectF.width() / 2;
        marginInfo.bottomMargin = bottomMargin+rectF.height()+offset;
    }
}

File: dathighlight/src/main/java/com/dat/dathighlight/position/OnTopPosCallback.kt
package com.dat.dathighlight.position

import android.graphics.RectF
import com.dat.dathighlight.HighLight.MarginInfo

/**
 * Được tạo bởi caizepeng vào 16/8/20.
 */
class OnTopPosCallback : OnBaseCallback {
    constructor()

    constructor(offset: Float) : super(offset)

    override fun getPosition(
        rightMargin: Float,
        bottomMargin: Float,
        rectF: RectF?,
        marginInfo: MarginInfo?
    ) {
        if (rectF == null || marginInfo == null) return

        marginInfo.leftMargin = rectF.right - rectF.width() / 2
        marginInfo.bottomMargin = bottomMargin + rectF.height() + offset
    }
}
